server/db/index.js

```diff
@@ -25,6 +25,29 @@ const get = (id) => {
   });
 };
 
-module.exports = { db, get };
+const insert = (data) => {
+  const entry = new Restaurants(data);
+  return entry.save();
+};
+
+const updateData = (id, data) => {
+    return new Promise((resolve, reject) => {
+      Restaurants.update({id}, data)
+        .exec((err, data) => {
+          if (err) {
+            reject(err);
+          }
+          resolve(data);
```

Comment:
You assigned native promises to `mongoose.Promise`, therefore, you do not need to build a promise. Instead `return Restaurants.update({id}, data).then(data => doSomething).catch(err => doSomething)`